src/plot_api/plot_schema.js

```diff
@@ -287,26 +277,37 @@ function getLayoutAttributes() {
 
             handleBasePlotModule(layoutAttributes, _module, astr);
         }
-    });
+    }
 
     // polar layout attributes
     layoutAttributes = assignPolarLayoutAttrs(layoutAttributes);
 
     // add registered components layout attributes
-    Object.keys(Registry.componentsRegistry).forEach(function(k) {
-        var _module = Registry.componentsRegistry[k];
-
-        if(!_module.layoutAttributes) return;
-
-        if(_module.schema && _module.schema.layout) {
-            Object.keys(_module.schema.layout).forEach(function(v) {
-                insertAttrs(layoutAttributes, _module.schema.layout[v], v);
-            });
+    for(key in Registry.componentsRegistry) {
+        _module = Registry.componentsRegistry[key];
+        var schema = _module.schema;
+
+        /*
+         * Components with defined schema have already been merged in at register time
+         * but a few components define attributes that apply only to xaxis
+         * not yaxis (rangeselector, rangeslider) - delete from y schema.
+         * Note that the input attributes for xaxis/yaxis are the same object
+         * so it's not possible to only add them to xaxis from the start.
+         * If we ever have such asymmetry the other way, or anywhere else,
+         * we will need to extend both this code and mergeComponentAttrsToSubplot
+         * (which will not find yaxis only for example)
+         */
+        if(schema && (schema.subplots || schema.layout)) {
+            var subplots = schema.subplots;
+            if(subplots && subplots.xaxis && !subplots.yaxis) {
```

Comment:
What if the `xaxis` and `yaxis` attributes were different objects?

Reply:
... would this simplify things?

Reply:
I'm thinking we _might_ have to make the `xaxis` and `yaxis` attribute objects different down the road if we (ever) want to have more details per-axis description (similar to https://github.com/plotly/plotly.js/issues/621)

Reply:
> What if the xaxis and yaxis attributes were different objects?

Then we'd need to extend this section 😈  Currently we don't have that so I didn't implement it, just added the comment above:
```
 * If we ever have such asymmetry the other way, or anywhere else,
 * we will need to extend both this code and mergeComponentAttrsToSubplot
 * (which will not find yaxis only for example)
```

Reply:
I suppose it would be useful to add an error check in there though, so it alerts us to fix it if we ever do add a y-axis-only, or different x and y, component attributes. I'll give that a shot.

Reply:
tested for y-only or x/y differences in https://github.com/plotly/plotly.js/pull/1999/commits/cb94e95d992c914faec1930da45b3a2a0af24b40